nexus/go-handler.md

```diff
@@ -0,0 +1,265 @@
+- Allow advanced users to implement Nexus inbound and outbound request proxies for cross-organization calls where
+  data converters are not shared and data is not exposed to Temporal server.
+
+  This will quickly become a requirement for production use cases, especially with Temporal Cloud's zero trust model. It
+  will be addressed shortly after Nexus becomes available for preview.
+
+> NOTE: These are non-goals for _this_ proposal, but they will be defined in follow up proposals / PRs.
+
+## Background
+
+Nexus is a thin protocol on top of HTTP, defined in [this spec][nexus-http-api]. Temporal serves as an HTTP proxy,
+translating Nexus requests into tasks that can be polled on by an SDK worker on its configured task queue.
+Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker
+to be processed.
+
+The process of translating requests into tasks and delivering them to workers is out of scope for this proposal.
+
+## SDK Additions
+
+The Temporal Go SDK exposes a `Worker` interface that embeds a `Registry` interface defined as:
+
+```go
+type Registry interface {
+	WorkflowRegistry
+	ActivityRegistry
+}
+```
+
+Add and embed a third registry type: `OperationRegistry` with the following method:
+
+```go
+import "github.com/nexus-rpc/sdk-go/nexus"
+
+type OperationRegistry interface {
+	RegisterOperation(operation string, handler nexus.Handler)
+}
+```
+
+> NOTE: `nexus.Handler` methods accept [request structs][nexus-start-request] that include an `*http.Request` object.
+> The SDK creates an fake request with the `Method`, `URL`, `Proto`, `Header`, and `Body` members populated (others
+> TBD).
+
+To minimize the amount of boilerplate required to map operations to common Temporal constructs, the SDK exposes default
+handler implementations.
+
+### Workflow Run
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type WorkflowStartOptions struct {
+	// Options to start the workflow with. TaskQueue defaults to the current worker's task queue.
+	// The workflow ID used here will be returned as the operation ID by default.
+	client.WorkflowStartOptions
+	// Type of workflow to start.
+	WorkflowType string
+	// Input to provide to the workflow.
+	Input        []any
+	// TODO: Should Temporal Header be settable here?
+}
+
+// Returns a Handler that can maps and operation to a workflow that can be started and canceled.
+// Handles HTTP request to payload deserialization for input of type `T`.
+func NewWorkflowRunHandler[T any](func(context.Context, T, *StartOperationRequest) (*WorkflowStartOptions, error)) Handler
+```
+
+**Basic usage:**
+
+```go
+w.RegisterOperation("provision-cell", NewWorkflowRunHandler(
+	func(ctx context.Context, input Input, request *StartOperationRequest) (*WorkflowStartOptions, error) {
+		return &WorkflowStartOptions{
+			Input:        []any{input},
+			WorkflowType: "provision-cell",
+			StartWorkflowOptions: StartWorkflowOptions{
```

Comment:
Having both `WorkflowStartOptions` and `StartWorkflowOptions` is going to be confusing.